libobs-opengl/gl-egl-common.c

```diff
@@ -199,24 +224,39 @@ gl_egl_create_dmabuf_image(EGLDisplay egl_display, unsigned int width,
 		return NULL;
 	}
 
-	if ((texture = gs_texture_create(width, height, color_format, 1, NULL,
-					 GS_DYNAMIC)) == NULL) {
+	texture = gl_egl_create_texture_from_eglimage(egl_display, width,
+						      height, color_format,
+						      GL_TEXTURE_2D, egl_image);
+	if (texture)
+		eglDestroyImage(egl_display, egl_image);
+
+	return texture;
+}
+
+struct gs_texture *
+gl_egl_create_texture_from_pixmap(EGLDisplay egl_display, uint32_t width,
+				  uint32_t height,
+				  enum gs_color_format color_format,
+				  EGLint target, EGLClientBuffer pixmap)
+{
+	if (!init_egl_image_target_texture_2d_ext())
 		return NULL;
-	}
-	const GLuint gltex = *(GLuint *)gs_texture_get_obj(texture);
 
-	gl_bind_texture(GL_TEXTURE_2D, gltex);
-	gl_tex_param_i(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-	gl_tex_param_i(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+	const EGLAttrib pixmap_attrs[] = {EGL_IMAGE_PRESERVED_KHR, EGL_TRUE,
+					  EGL_NONE};
```

Comment:
Nit: does clang complain if this array is declared as:

```c
const EGLAttrib pixmap_attrs[] = {
	EGL_IMAGE_PRESERVED_KHR,
	EGL_TRUE,
	EGL_NONE,
};
```

?

Reply:
I think this is how clang-format 12.0.0 will always format those.

Reply:
Clang-format just leaves this ambiguous. if you dont put an element on the line with the curly brace it should preserve the one element per line format.